jni/src/native_channel.c

```diff
@@ -123,6 +126,11 @@ JNIEXPORT jint JNICALL Java_tel_schich_libdatachannel_LibDataChannelNative_rtcSe
     if (data == NULL) {
         return RTC_ERR_SUCCESS;
     }
+    jlong bufferSize = (*env)->GetDirectBufferCapacity(env, data);
```

Comment:
I'd prefer to do this validation in java-land and trust that this function is not called with non-sense parameters.